tools/clang/test/CodeGenDXIL/literal/fmod_const_eval.hlsl

```diff
@@ -0,0 +1,43 @@
+// RUN: %dxc -T vs_6_0 %s -E main | FileCheck %s
+// RUN: not %dxc -T vs_6_0 %s -E main -DNO_FOLD 2>&1 | FileCheck %s --check-prefixes=NO_FOLD
+
+// The code path for this case asserts, but the assert does not indicate a
+// serious problem with internal state.  The invalid overload will either be
+// constant folded away, or caught by the validator.
+
+// Ensure fmod is constant evaluated during codegen, or dxil const eval
+// TODO: handle fp specials properly!
```

Comment:
The fact that we produce an invalid overload with user-writable code is troubling.  Is this TODO about fixing that? The comments in #6410 suggested that we would keep it open to fix the root cause even after this partial revert. If that's the case, it should be cited here and the "fixes" line should be removed from the description. If not, we need another bug to cite here.

Reply:
The fp specials issue is tracked with #6567.

Reply:
I think we decided to punt on fixes, we need to reference the bugs in the code. I'd like to see it mentioend in the comments.